kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -108,24 +122,19 @@ type MatchOverloadingResult unifier variable =
         unifier
         (Pair (TermLike variable))
 
-type OverloadingResult unifier a = ExceptT UnifyOverloadingError unifier a
-
-notApplicable :: Monad unifier => OverloadingResult unifier a
-notApplicable = empty
-
-throwBottom :: Monad unifier => String -> OverloadingResult unifier a
-throwBottom = throwE . Clash
+throwBottom :: String -> Maybe MatchResult
+throwBottom = Just . ClashResult
 
 matchOverloading ::
     MonadSimplify unifier =>
     Pair (TermLike RewritingVariableName) ->
     MatchOverloadingResult unifier RewritingVariableName
-matchOverloading termPair =
-    do
-        unifyResult <- unifyOverloading termPair
-        case unifyResult of
-            Simple pair -> return pair
-            _ -> notApplicable
+matchOverloading termPair = do
+    overloadSimplifier <- askOverloadSimplifier
+    case unifyOverloading overloadSimplifier termPair of
+        Just (Resolution (Simple pair)) -> return pair
+        Just (ClashResult msg) -> throwE $ Clash msg
```

Comment:
What happens when they `Clash`? Can you point out where we call this or handle this case? Sorry, I wasn't really familiar with this part of the code before, so I'm trying to understand the big picture here.

Reply:
Not sure if this is what you're asking but `matchOverloading` is only called directly [here](https://github.com/kframework/kore/blob/67d4d23698551ba29228864fdff55a15d86f3d55/kore/src/Kore/Step/Axiom/Matcher.hs#L426) and [here](https://github.com/kframework/kore/blob/e0614507f2172e8d161fa8e8eddd844dba724eae/kore/test/Test/Kore/Step/Simplification/Overloading.hs#L519).